cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/reserve_transfer.rs

```diff
@@ -1132,8 +1145,666 @@ fn reserve_transfer_native_asset_from_para_to_para_through_relay() {
+}
+
+// ================================================================================
+// ===== Reserve Transfers - Sufficient Asset - Parachain->AssetHub->Parachain ====
+// ================================================================================
+/// Reserve Transfers of a sufficient trust-backed asset from Parachain to Parachain (through Relay
+/// reserve) should work
+#[test]
+fn reserve_transfer_sufficient_asset_from_para_to_para_through_asset_hub() {
+	use penpal_runtime::xcm_config::ASSET_HUB_ID;
+
+	// Init values for Parachain Origin
+	let destination = PenpalA::sibling_location_of(PenpalB::para_id());
+	let sender = PenpalASender::get();
+	let asset_amount_to_send: Balance = WESTEND_ED * 10000;
+	let fee_amount_to_send: Balance = WESTEND_ED * 10000;
+	let asset_owner = PenpalAssetOwner::get();
+	let sender_as_seen_by_asset_hub = AssetHubWestend::sibling_location_of(PenpalA::para_id());
+	let sov_of_sender_on_asset_hub =
+		AssetHubWestend::sovereign_account_id_of(sender_as_seen_by_asset_hub);
+
+	// Create sufficient asset.
+	let asset_id = 1984;
+	AssetHubWestend::force_create_asset(
+		asset_id,
+		asset_owner,
+		true, // Mark it as sufficient.
+		70_000,
+		vec![(sov_of_sender_on_asset_hub.clone(), asset_amount_to_send + fee_amount_to_send)],
+	);
+	let system_para_foreign_asset_location = Location::new(
+		1,
+		[Parachain(ASSET_HUB_ID), PalletInstance(ASSETS_PALLET_ID), GeneralIndex(asset_id.into())],
+	);
+
+	// Create sufficient foreign asset.
+	PenpalA::force_create_foreign_asset(
+		system_para_foreign_asset_location.clone(),
+		PenpalAssetOwner::get(),
+		false, /* TODO: Doesn't matter if sufficient since we don't handle local delivery fee
+		        * payment in other assets. */
+		70_000,
+		vec![(sender.clone(), asset_amount_to_send + fee_amount_to_send)],
+	);
+	PenpalB::force_create_foreign_asset(
+		system_para_foreign_asset_location.clone(),
+		PenpalAssetOwner::get(),
+		true, // Mark as sufficient.
+		70_000,
+		vec![],
+	);
+
+	let assets: Assets = vec![(
+		(
+			Parent,
+			Parachain(ASSET_HUB_ID),
+			PalletInstance(ASSETS_PALLET_ID),
+			GeneralIndex(asset_id.into()),
+		),
+		asset_amount_to_send + fee_amount_to_send,
+	)
+		.into()]
+	.into();
+	// Just to be very specific you don't need any native asset.
+	assert_eq!(assets.len(), 1);
+
+	// fund the Parachain Origin's SA on AssetHub with the native tokens held in reserve
+	AssetHubWestend::fund_accounts(vec![(
+		sov_of_sender_on_asset_hub.into(),
+		asset_amount_to_send * 2,
+	)]);
+
+	// Give the sender enough Relay tokens to pay for local delivery fees.
+	// TODO: When we support local delivery fee payment in other assets, we don't need this.
+	PenpalA::execute_with(|| {
+		assert_ok!(<PenpalA as PenpalAPallet>::ForeignAssets::mint(
+			<PenpalA as Chain>::RuntimeOrigin::signed(PenpalAssetOwner::get()),
+			Location::new(1, []),
+			sender.clone().into(),
+			10_000_000_000_000, // Big estimate to make it work.
+		));
+	});
```

Comment:
You can just fund pallet-balances (PENs) here for local fees (more realistic)

Reply:
I tried but it didn't work. Penpal's XCM config is set up to pay fees in the relay token (WND)

Reply:
ah, interesting.. that's unfortunate because it's not representative of ecosystem parachains

can you change that to use PENs in a separate PR (to not blow up this one)? we will need PEN/WND pool on Penpal for execution fees for existing tests, but afaik we don't need this PR too - it can come on top with the new scenarios where delivery fees are also paid using other stuff

Reply:
Sounds good, I thought about changing it here, but it changed a lot of tests 😅